test/validation/dry_validation_test.rb

```diff
@@ -363,6 +363,21 @@ def good_musical_taste?(value)
       form.producers.first.errors.messages.inspect.must_equal %({:name=>[\"must be filled\"]})
       form.errors.messages.inspect.must_equal %({:title=>["must be filled", "you're a bad person"], :"band.name"=>["must be filled"], :"band.label.name"=>["must be filled"], :"producers.name"=>[\"must be filled\"], :"hit.title"=>["must be filled"], :"songs.title"=>["must be filled"]})
     end
+
+    it "renders full messages correcty" do
+      result = form.validate(
+        "title"  => "",
+        "songs"  => [ {"title" => ""}, {"title" => ""} ],
+        "band"   => {"size" => "", "label" => {"name" => ""}},
+        "producers" => [{"name" => ''}, {"name" => 'something lovely'}]
+      )
+
+      result.must_equal false
+      form.band.errors.full_messages.inspect.must_equal %({:name=>["Name must be filled"], :\"label.name\"=>[\"Label Name must be filled\"]})
```

Comment:
Thanks for pointing this out. I'll confess that I've never actually used ActiveRecords full messages and couldn't remember exactly how they came out. 

Does this look better? https://github.com/trailblazer/reform/tree/fix-arstyle-full-messages